code/__HELPERS/turfs.dm

```diff
@@ -279,6 +279,76 @@ Turf and target are separate in case you want to teleport some distance from a t
 	LAZYSET(modifiers, ICON_Y, "[(click_turf_py - click_turf.pixel_y) + ((click_turf_y - click_turf.y) * ICON_SIZE_Y)]")
 	return click_turf
 
+/**
+ * Converts mouse-pos control coordinates to a specific turf location on the map.
+ *
+ * Handles the conversion between control-space mouse coordinates and screen-space map coordinates,
+ * accounting for various BYOND quirks and display scaling factors.
+ *
+ * @param mousepos_x	The x-coordinate of the mouse click (in control pixels, top-left origin)
+ * @param mousepos_y	The y-coordinate of the mouse click (in control pixels, top-left origin)
+ * @param sizex			x control width of the map
+ * @param sizey			y control width of the map
+ * @param viewing_client The client whose view perspective to use for the conversion
+ *
+ * @return The turf at the calculated map position, or the closest one if out of bounds, as well as the residual x and y map offsets
+ *
+ * Important Notes:
+  * - This WILL be incorrect when client pixel_wxyz is animating, and it WILL be incorrect if the user is gliding, because we don't have a good way to compensate this on the serverside. Yay!!!
+ * - Mouse coordinates originate from the top-left corner because we can't have consistency in this engine
+ * - Coordinate systems are inconsistent between control pixels and screen pixels
+ * - Something on the byond side (icon size likely? needs debugging) affects the control pixels
+ * - This uses the ratios between them rather than absolute values for reliable results
```

Comment:
```suggestion
 *
 * Important Notes:
 * - This WILL be incorrect when client pixel_wxyz is animating, and it WILL be incorrect if the user is gliding, because we don't have a good way to compensate this on the serverside. Yay!!!
 * - Mouse coordinates originate from the top-left corner because we can't have consistency in this engine
 * - Coordinate systems are inconsistent between control pixels and screen pixels
 * - Something on the byond side (icon size likely? needs debugging) affects the control pixels
 * - This uses the ratios between them rather than absolute values for reliable results
```